### PR TITLE
add downstream tests for awslabs/aws-dynamodb-encryption-python

### PR DESCRIPTION
**Pull request by mattsb42-aws**
Add downstream tests for https://github.com/awslabs/aws-dynamodb-encryption-python

### CODE REVIEW

File: .travis/run.sh [hunk 1]
```diff
@@ -54,6 +54,13 @@ else
             pip install -e .
             pytest -m local -l
             ;;
+        dynamodb-encryption-sdk)
+            git clone --depth=1 https://github.com/awslabs/aws-dynamodb-encryption-python
+            cd aws-dynamodb-encryption-python
+            pip install -r test/requirements.txt
+            pip install -e .
+            pytest -m "local and not slow and not veryslow and not nope" -l
```
Comment (alex) [inline .travis/run.sh]: I lol'd

**Comment by alex** [comment]
The tests seem to be failing 👀

**Comment by mattsb42-aws** [comment]
Yeah, was just digging through the logs (side note, might be worth revisiting the whitelisted branches? makes it harder to run the CI prior to a PR). Looks like it's choking on RSA-OAEP-SHA256, which requires OpenSSL >= 1.0.2.

**Comment by alex** [comment]
Oh gosh, it makes me so happy that our new downstream test harness makes it easy to swap OpenSSLs

File: .travis/run.sh (hunk 1 above)
Comment (mattsb42-aws) [reply .travis/run.sh]: Haha, yup. Of the many levels of hypothesis-powered combinatorically exploding test scenarios, I figure we want the smallest set here.

**Comment by reaperhulk** [comment]
@mattsb42-aws for future reference we're totally okay with iterating a bit in a PR. A few build failures is fine. 😄 We have more parallel builders than your fork too so the jobs will run faster anyway.

**Comment by mattsb42-aws** [comment]
@reaperhulk Fair enough. I was under the impression that Travis ran fork CI in the context of the fork owner, so the builds for a fork wouldn't affect the main repo CI.

**Comment by alex** [comment]
@mattsb42-aws I cannot begin to tell you how much that wouldn't bother us :-) FWIW, we're not going to remove the branch restriction because otherwise every time I send a PR via the github edit-in-the-web UI, the tests get run twice and it makes @reaperhulk sad.

**Comment by mattsb42-aws** [comment]
@alex Haha, fair enough. I guess I could always just make an ephemeral branch on my fork that adds itself to the whitelist and gets constantly rebased on top of any branch I'm actually working on. Would be nice if the whitelist controls could be set per fork as well as per branch.

**Pull request merged**